src/meltano/core/tracking/tracker.py

```diff
@@ -0,0 +1,196 @@
+
+        if emitters:
+            self.snowplow_tracker = SnowplowTracker(emitters=emitters)
+            self.snowplow_tracker.subject.set_lang(locale.getdefaultlocale()[0])
+            self.snowplow_tracker.subject.set_timezone(self.timezone_name)
+        else:
+            self.snowplow_tracker = None
+
+        self.send_anonymous_usage_stats = self.settings_service.get(
+            "send_anonymous_usage_stats", True
+        )
+
+        self.contexts: tuple[SelfDescribingJson] = (
+            environment_context,
+            ProjectContext(project),
+        )
+
+    @cached_property
+    def timezone_name(self) -> str:
+        """Obtain the local timezone name.
+
+        Examples:
+            The timezone name as an IANA timezone database name:
+
+                >>> SnowplowTracker(project).timezone_name
+                'Europe/Berlin'
+
+            The timezone name as an IANA timezone abbreviation because the full name was not found:
+
+                >>> SnowplowTracker(project).timezone_name
+                'CET'
+
+        Returns:
+            The local timezone as an IANA TZ database name if possible, or abbreviation otherwise.
+        """
+        try:
+            return tzlocal.get_localzone_name()
+        except Exception:
+            return datetime.datetime.now().astimezone().tzname()
+
+    @contextmanager
+    def with_contexts(self, *extra_contexts) -> Tracker:
+        """Context manager within which the `Tracker` has additional Snowplow contexts.
+
+        Args:
+            extra_contexts: The additional contexts to add to the `Tracker`.
+
+        Yields:
+            A `Tracker` with additional Snowplow contexts.
+        """
+        prev_contexts = self.contexts
+        self.contexts = (*prev_contexts, *extra_contexts)
+        try:
+            yield self
+        finally:
+            self.contexts = prev_contexts
+
+    def can_track(self) -> bool:
+        """Check if the tracker can be used.
+
+        Returns:
+            True if the tracker can be used, False otherwise.
+        """
+        return self.snowplow_tracker is not None and self.send_anonymous_usage_stats
+
+    def track_struct_event(self, category: str, action: str) -> None:
+        """Fire a structured tracking event.
+
+        Note: This is a legacy method that will be removed in a future version, once LegacyTracker is no longer used.
+
+        Args:
+            category: The category of the event.
+            action: The event actions.
+        """
+        if not self.can_track():
+            return
+
+        if self.project.active_environment is not None:
+            action = f"{action} --environment={hash_sha256(self.project.active_environment.name)}"
+
+        try:
+            self.snowplow_tracker.track_struct_event(
+                category=category,
+                action=action,
+                label=self.project_id,
+            )
+        except Exception as ex:
+            logger.warning(
```

Comment:
@aaronsteers I think im gonna switch these to logger.debug. If we run into issues with telemetry I don't wanna spam the end-user with errors, and we can easily enough troubleshoot using --log-level=debug. wdyt ?

Reply:
Yeah - makes sense! 👍